#! /bin/sh -e

if [ $# -eq 3 -a "$2" = '-d' ]; then
    pdir="-d $3"
elif [ $# -ne 1 ]; then
    echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
    exit 1
fi
case "$1" in
    -patch) patch $pdir -f --no-backup-if-mismatch -p0 < $0;;
    -unpatch) patch $pdir -f --no-backup-if-mismatch -R -p0 < $0;;
    *)
	echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
	exit 1
esac
exit 0

# DP: Don't mishandle negative pod in `kill' builtin

--- builtins/kill.def~	Mon Oct 15 19:23:32 2001
+++ builtins/kill.def	Sat Apr  6 21:12:55 2002
@@ -163,12 +163,12 @@
       if (*word == '-')
 	word++;
 
+      /* Use the entire argument in case of minus sign presence. */
       if (*word && legal_number (word, &pid_value) && (pid_value == (pid_t)pid_value))
 	{
-	  /* Use the entire argument in case of minus sign presence. */
 	  pid = (pid_t) pid_value;
 
-	  if (kill_pid (pid, signal, 0) < 0)
+	  if ((pid < -1 ? kill_pid (-pid, signal, 1) : kill_pid (pid, signal, 0)) < 0)
 	    goto signal_error;
 	  else
 	    any_succeeded++;
